### ISSUE TITLE
ChromeDriver source link is broken

### ISSUE THREAD

**Issue by opemvbs** [ultrafunkamsterdam/undetected-chromedriver#1743]
I have a program that uses uc, today after restarting it (the program has been run for like 7 hours), I got HTTP error 404 from requests module, but after searching it more, I bet the 404 is because the chromedriver source link that's getting downloaded to be patched (in patcher.py) is broken, I tried copy and pasting the link `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing` to be manually accessed, and yeah, google gives 404 error.
    
```
    def fetch_package(self):
        """
        Downloads ChromeDriver from source

        :return: path to downloaded file
        """
        zip_name = f"chromedriver_{self.platform_name}.zip"
        if self.is_old_chromedriver:
            download_url = "%s/%s/%s" % (self.url_repo, self.version_full.vstring, zip_name)
        else:
            zip_name = zip_name.replace("_", "-", 1)
            download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"
            download_url %= (self.version_full.vstring, self.platform_name, zip_name)

        logger.debug("downloading from %s" % download_url)
        return urlretrieve(download_url)[0]
```

**Comment by opemvbs** [comment]
I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.

`download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"`
to
`download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

**Comment by omkmorendha** [comment]
This is a command you can use after going to the (virtual env path)/lib/python3.xx/site-packages/undetected_chromedriver directory

`sed -i "s|download_url = \"https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s\"|download_url = \"https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s\"|g" patcher.py `

**Comment by Nygosaki** [comment]
Same issue, @ultrafunkamsterdam would you mind changing the URL?

**Comment by Nygosaki** [comment]
> This is a command you can use after going to the (virtual env path)/lib/python3.xx/site-packages/undetected_chromedriver directory
> 
> `sed -i "s|download_url = \"https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s\"|download_url = \"https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s\"|g" patcher.py `

I am getting `sed: 1: "patcher.py": extra characters at the end of p command`

**Comment by espoir911** [comment]
what does this url contain, is it possible to download it locally and open it locally to make sure the problem doesn't recur?

**Comment by mshot** [comment]
> what does this url contain, is it possible to download it locally and open it locally to make sure the problem doesn't recur?

binaries for testing - https://storage.googleapis.com/chrome-for-testing-public/121.0.6167.184/win64/chrome-win64.zip

**Comment by ifeldshteyn** [comment]
> I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.
> 
> `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"` to `download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

Confirmed this fixed it for me. Can we disable patching in general?

**Comment by Martini002** [comment]
> I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.
> 
> `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"` to `download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

confirmed, worked for me too, can someone explain us what the patcher.py does, is it really needed?
thank you

**Comment by charlbury** [comment]
> I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.
> 
> `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"` to `download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

worked for me

**Comment by aaqibmehrban** [comment]
> I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.
> 
> `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"` to `download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

I had issue on 2 pcs, for one it worked and for other when i restarted, it automatically worked without needing to edit patcher.py

**Comment by ror74559** [comment]
Thanks, it worked for me.

**Comment by hemalsimon** [comment]
I've been facing this issue in the Ubuntu environment. Do you know how I can fix this issue? 
![Screenshot 2024-02-17 124102](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/49345867/78c76fe1-ab70-4430-88f8-2f985dd2ba89)

**Comment by julien-fr** [comment]
> This is a command you can use after going to the (virtual env path)/lib/python3.xx/site-packages/undetected_chromedriver directory
> 
> `sed -i "s|download_url = \"https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s\"|download_url = \"https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s\"|g" patcher.py `

Fastest working solution! Thanks.

**Comment by xzcrrazykidzx** [comment]
> I temporarily fixed in my device by changing a line in your local undetected_chromedriver installation patcher.py file, but I think it'll work just fine on other people too.
> 
> `download_url = "https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/%s"` to `download_url = "https://storage.googleapis.com/chrome-for-testing-public/%s/%s/%s"`

not working for me, for this error after replacing that line:
<urlopen error [WinError 10054] An existing connection was forcibly closed by the remote host>

**Comment by ilike2burnthing** [comment]
Issue resolved in v3.5.5
